pkg/services/sqlstore/temp_user.go

```diff
@@ -132,3 +133,17 @@ func GetTempUserByCode(query *models.GetTempUserByCodeQuery) error {
 	query.Result = &tempUser
 	return err
 }
+
+func ExpireOldUserInvites(cmd *models.ExpireTempUsersCommand) error {
+	return inTransaction(func(sess *DBSession) error {
+		var rawSql = "UPDATE temp_user SET status = ? WHERE created <= ? AND status != ?"
+		expiredStatus := string(models.TmpUserExpired)
```

Comment:
There's a potential timezone issue here when comparing date type (created) in database query. We had several bugs related to that throughout the years and the proper solution has been to add a new column holding an epoch value. I may be wrong, but think it should be possible to verify this with a test and mocking the time and then integration tests should catch it. For reference #10947 #10904 #10907 as examples where we fixed this earlier. 

It's tricky though since you normally have to drop the table and create a new one or add one additional column. Need to verify if this a problem first. Could be that our integration tests don't catch these kind of errors.

However, I suggest updating invites that are in pending state. WDYT?

```suggestion
		var rawSql = "UPDATE temp_user SET status = ? WHERE created <= ? AND status = ?"
		expiredStatus := string(models.TmpUserInvitePending)
```

Reply:
Ah good catch. Let me test it out and report back!

Regarding the state check, yep that makes sense since this will even change successfully completed invites for example which we don't want. With that in mind, we're okay with leaving invites in a state of `SignUpStarted` indefinitely?

Reply:
Putting here for completeness:

Results show that there is indeed differing results between the 3 different SQL sources. Here's the test:

```
Convey("Should respect timezone differences when expiring invites ", func() {
    now := time.Date(2020, 1, 1, 12, 0, 0, 0, time.UTC) // 1 Jan 2020 12:00:00 UTC
    timeNow = func() time.Time { return now }
    cmd := models.CreateTempUserCommand{
        OrgId:  123,
        Name:   "test",
        Code:   "789",
        Email:  "test@grafana.com",
        Status: models.TmpUserInvitePending,
    }
    err := CreateTempUser(&cmd)
    So(err, ShouldBeNil)
    tmpUser := cmd.Result
    So(tmpUser.Created, ShouldEqual, now)
    utcPlus1 := time.FixedZone("UTC+1", 1*60*60)
    olderThan := time.Date(2020, 1, 1, 12, 0, 0, 0, utcPlus1) // 1 Jan 2020 12:00:00 UTC+1 (IE 1 Jan 2020 11:00:00 UTC)
    cmd4 := models.ExpireTempUsersCommand{OlderThan: olderThan}
    err = ExpireOldUserInvites(&cmd4)
    So(err, ShouldBeNil)
    So(cmd4.NumExpired, ShouldEqual, 0)
    olderThan = time.Date(2020, 1, 1, 13, 0, 0, 0, utcPlus1) // 1 Jan 2020 13:00:00 UTC+1 (IE 1 Jan 2020 12:00:00 UTC)
    cmd4 = models.ExpireTempUsersCommand{OlderThan: olderThan}
    err = ExpireOldUserInvites(&cmd4)
    So(err, ShouldBeNil)
    So(cmd4.NumExpired, ShouldEqual, 1)
})
```
Passes on MySQL, but fails on SQLite and PostgreSQL

The plan now is to look at a migration so that we deal exclusively with epoch similar to other migrations done in the past

Reply:
Great :+1: